client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -157,10 +160,7 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 			const apiVersion = siteIsJetpack ? jetpackSiteSettingsAPIVersion : '1.4';
 			this.props.saveSiteSettings( siteId, { ...siteFields, apiVersion } );
 			if ( jetpackSettingsUISupported ) {
-				const fieldsToUpdate = /^error_/.test( fields.lang_id )
-					? omit( fields, 'lang_id' )
-					: fields;
-				this.props.updateSettings( siteId, pick( fieldsToUpdate, settingsFields.jetpack ) );
+				this.props.saveJetpackSettings( siteId, jetpackFieldsToUpdate );
```

Comment:
❌ It seems like there is some regression when there is an error in saving. To reproduce:

1. Go to http://calypso.localhost:3000/settings/traffic/:site
2. Make sure Analytics module is enabled.
3. Try saving the Analytics card with an empty "Google Analytics Tracking ID" field.
4. In production you will see an error notice, but in this branch nothing happens.

Reply:
That's probably because https://github.com/Automattic/wp-calypso/pull/23797/files#diff-94ac3f3b3f86928d526e5d90324a2599R73 is too generic after all (don't show these error notices if this settings panel includes JP settings, since the JP settings data layer displays those notices for us). I should probably replace it with a check for the result of `isJetpackSettingsSaveFailure` (don't show these error notices if a JP settings save action has failed, since the JP settings data layer displays those notices for us).

Reply:
Should be fixed by f84750e.